File: src/main/java/com/github/mystery2099/Voxlib.java
package com.github.mystery2099;

import net.neoforged.bus.api.IEventBus;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.ModContainer;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.fml.common.Mod;
import net.neoforged.fml.event.lifecycle.FMLCommonSetupEvent;
import net.neoforged.neoforge.common.NeoForge;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@EventBusSubscriber
@Mod("voxlib")
public class Voxlib {
    private static final String MODID = "voxlib";
    private static final Logger logger = LoggerFactory.getLogger(MODID);

    public Voxlib(IEventBus modEventBus, ModContainer modContainer) {}

    @SubscribeEvent
    public static void onCommonSetup(FMLCommonSetupEvent event) {
        logger.debug("Voxlib initialized!");
    }
}

File: src/main/kotlin/com/github/mystery2099/voxlib/VoxLib.kt
package com.github.mystery2099.voxlib

import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.fml.common.EventBusSubscriber
import net.neoforged.fml.common.Mod
import org.slf4j.LoggerFactory

/**
 * Main entry point for the VoxLib mod.
 * This library provides utilities for manipulating, creating, and rotating voxel shapes.
 */
@Mod("voxlib")
@EventBusSubscriber
object VoxLib {
    private val logger = LoggerFactory.getLogger("voxlib")
    private const val VERSION = "1.2.0-neo"
    private const val MINECRAFT_VERSION = "1.21.1"

    @SubscribeEvent
	@JvmStatic fun log() {
        logger.info("Initializing VoxLib for $MINECRAFT_VERSION")
    }
}